src/wurf/dependency_manager.py

```diff
@@ -63,13 +90,21 @@ def add_dependency(self, **kwargs):
         if not path:
             return
 
+        print("RECURSE {}".format(dependency.recurse))
+
         self.dependency_cache[dependency.name] = \
             {'path': path, 'recurse': dependency.recurse}
 
         if dependency.recurse:
             # We do not require the 'resolve' function to be implemented in
             # dependency projects. Therefore the mandatory=False.
-            self.ctx.recurse(path, mandatory=False)
+            #
+            # @todo the str() here is needed as waf does not handle unicode
+            # in its find_node function. So that would be nice to fix.
+            self.ctx.recurse(str(path), mandatory=False)
```

Comment:
I think the first argument for recurse should always be `[str(path)]`, then we can also support paths that contain spaces. Maybe we should create a dedicated test for these fancy paths.